src/services/APIServiceBase.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * This is the base class for all services that interact with the API system.
+ */
+
+import { Validation } from "io-ts";
+import * as msRest from "ms-rest-js";
+import * as winston from "winston";
+import { ReadableReporter } from "../utils/validation_reporters";
+
+const messageErrorOnUnknownResponse = "Unknown response.";
+const messageErrorOnApiError = "Api error.";
+
+export default class APIServiceBase {
+  protected unknownResponseError(
+    // tslint:disable-next-line:no-any
+    error: Validation<any>,
+    service: string
+  ): Error {
+    winston.error(
+      "Unknown response from %s API: %s",
+      service,
+      ReadableReporter.report(error)
+    );
+    return new Error(messageErrorOnUnknownResponse);
+  }
+
+  // tslint:disable-next-line:no-any
+  protected apiError(error: Validation<any>, service: string): Error {
+    winston.error("API error in %s: %s", service, error);
+    return new Error(messageErrorOnApiError);
+  }
+
+  protected extractBodyFromResponse(
```

Comment:
I see most times you will call both `extractBodyFromResponse` and `extractStatusFromResponse`, it could make sense to have either one single method that returns both values (e.g. a tuple) or a wrapper on the `msRest` object:

```ts
function getOpResponseBodyAndStatus(httpOperationResponse: msRest.HttpOperationResponse): [any, number] {
}
```

```ts
class SimpleHttpOperationResponse {
  constructor(private readonly httpOperationResponse: msRest.HttpOperationResponse) {}
  bodyAsJson(): any { return this.httpOperationResponse.bodyAsJson }
  status(): number { return this.httpOperationResponse.response.status }
}
```